src/lib/mdns/Advertiser_ImplMinimalMdns.cpp

```diff
@@ -209,36 +216,80 @@ CHIP_ERROR AdvertiserMinMdns::Start(chip::Inet::InetLayer * inetLayer, uint16_t
 /// Stops the advertiser.
 CHIP_ERROR AdvertiserMinMdns::StopPublishDevice()
 {
-    mQueryResponderAllocatorOperational.Clear();
+    for (size_t i = 0; i < kMaxOperationalNetworks; ++i)
+    {
+        mQueryResponderAllocatorOperational[i].Clear();
+    }
     mQueryResponderAllocatorCommissionable.Clear();
     mQueryResponderAllocatorCommissioner.Clear();
     return CHIP_NO_ERROR;
 }
 
+int AdvertiserMinMdns::FindOperationalAllocator(const FullQName & qname) const
+{
+    for (size_t i = 0; i < kMaxOperationalNetworks; ++i)
+    {
+        if (mQueryResponderAllocatorOperational[i].GetResponder(QType::SRV, qname) != nullptr)
+        {
+            return i;
+        }
+    }
+    return -1;
+}
+
+int AdvertiserMinMdns::FindEmptyOperationalAllocator() const
+{
+    for (size_t i = 0; i < kMaxOperationalNetworks; ++i)
+    {
+        if (mQueryResponderAllocatorOperational[i].IsEmpty())
+        {
+            return i;
+        }
+    }
+    return -1;
+}
+
 CHIP_ERROR AdvertiserMinMdns::Advertise(const OperationalAdvertisingParameters & params)
 {
-    // TODO: When multi-admin is enabled, commissionable does not need to be cleared here.
-    mQueryResponderAllocatorOperational.Clear();
-    char nameBuffer[64] = "";
+    char nameBuffer[kOperationalServiceNamePrefix + 1] = "";
 
     /// need to set server name
     ReturnErrorOnFailure(MakeInstanceName(nameBuffer, sizeof(nameBuffer), params.GetPeerId()));
 
+    char nameCheckStorage[kMaxOperationalServiceNameSize + 4 * sizeof(QNamePart)];
+    FullQName nameCheck =
+        FlatAllocatedQName::Build(nameCheckStorage, nameBuffer, kOperationalServiceName, kOperationalProtocol, kLocalDomain);
+    int idx = FindOperationalAllocator(nameCheck);
```

Comment:
And here we'd do:
```suggestion
    auto * allocator = FindOperationalAllocator(nameCheck);
```
and then null-check it, etc.